test/truefi2/lines-of-credit/TrueCreditAgency.test.ts

```diff
@@ -1064,80 +1046,43 @@ describe('TrueCreditAgency', () => {
         const tx = await pendingTx
         const receipt = await tx.wait()
         const iface = loanFactory.interface
-        return DebtToken__factory.connect(
-          receipt.events
-            .filter(({ address }) => address === loanFactory.address)
-            .map((e) => iface.parseLog(e))
-            .find(({ eventFragment }) => eventFragment.name === 'DebtTokenCreated')
-            .args.contractAddress,
-          owner)
+        return Promise.all(receipt.events
+          .filter(({ address }) => address === loanFactory.address)
+          .map((e) => iface.parseLog(e))
+          .filter(({ eventFragment }) => eventFragment.name === 'DebtTokenCreated')
+          .map((e) => DebtToken__factory.connect(e.args.contractAddress, owner)))
       }
 
       it('creates DebtToken with expected params', async () => {
-        const debtToken = await extractDebtTokenAddress(creditAgency.enterDefault(tusdPool.address, borrower.address))
-        expect(await debtToken.pool()).to.eq(tusdPool.address)
-        expect(await debtToken.borrower()).to.eq(borrower.address)
-        expect(await debtToken.debt()).to.eq(1000)
+        const debtTokens = await extractDebtTokens(creditAgency.enterDefault(borrower.address))
+        expect(await debtTokens[1].pool()).to.eq(tusdPool.address)
+        expect(await debtTokens[1].borrower()).to.eq(borrower.address)
+        expect(await debtTokens[1].debt()).to.eq(1000)
       })
 
       it('creates multiple DebtTokens for different pools', async () => {
-        const tx = await creditAgency.enterDefault(tusdPool.address, borrower.address)
-        expect(tx).to.emit(loanFactory, 'DebtTokenCreated')
-        const tx2 = await creditAgency.enterDefault(usdcPool.address, borrower.address)
-        expect(tx2).to.emit(loanFactory, 'DebtTokenCreated')
+        const debtTokens = await extractDebtTokens(creditAgency.enterDefault(borrower.address))
+        expect(debtTokens.length).to.eq(2)
+        expect(await debtTokens[0].pool()).to.eq(usdcPool.address)
+        expect(await debtTokens[1].pool()).to.eq(tusdPool.address)
       })
 
-      it('unlocks borrowing mutex after all loans were defaulted', async () => {
-        await creditAgency.enterDefault(tusdPool.address, borrower.address)
-        expect(await borrowingMutex.locker(borrower.address))
-          .to.equal(creditAgency.address)
-        await creditAgency.enterDefault(usdcPool.address, borrower.address)
-        expect(await borrowingMutex.locker(borrower.address))
-          .to.equal(AddressZero)
+      it('only creates DebtTokens for pools with nonzero debt', async () => {
+        await creditAgency.connect(borrower).repayInFull(usdcPool.address)
+        const debtTokens = await extractDebtTokens(creditAgency.enterDefault(borrower.address))
+        expect(debtTokens.length).to.eq(1)
+        expect(await debtTokens[0].pool()).to.eq(tusdPool.address)
       })
-    })
-  })
 
-  describe('pokeBorrowLimitTimer', () => {
```

Comment:
If you expand the code below, you'll see this section has been duplicated (probably due to merge conflict resolutions).